commands/utils.go

```diff
@@ -184,21 +199,41 @@ func buildReport(results []Result) error {
 		}
 
 		if rs.Output != nil {
-			fmt.Printf("node[%d] exit %d\n", rs.Node, rs.Output.ExitCode())
-			if rs.Output.Error() != nil {
-				fmt.Printf("%s", rs.Output.Error())
+			if encoding == "text" {
+				fmt.Printf("node[%d] exit %d elapsed %s\n", rs.Node, rs.Output.ExitCode(), rs.Elapsed)
+				if rs.Output.Error() != nil {
+					fmt.Printf("%s", rs.Output.Error())
+				}
+				io.Copy(os.Stdout, rs.Output.Stdout())
+				io.Copy(os.Stdout, rs.Output.Stderr())
+			} else {
+				// Transform plugin stdout to string
+				pluginOutB, err := ioutil.ReadAll(rs.Output.Stdout())
+				if err != nil {
+					errs = append(errs, err)
+				}
+				pluginOut := string(pluginOutB)
+				// Transform plugin stderr to string
+				pluginErrB, err := ioutil.ReadAll(rs.Output.Stderr())
+				if err != nil {
+					errs = append(errs, err)
+				}
+				pluginErr := string(pluginErrB)
+				// JSONify the plugin output
+				rsJSON, _ := json.Marshal(output{
+					Node:         rs.Node,
+					ExitCode:     rs.Output.ExitCode(),
+					Error:        rs.Output.Error(),
+					PluginStdout: pluginOut,
+					PluginStderr: pluginErr,
+					Elapsed:      rs.Elapsed.Seconds(),
+				})
+				fmt.Printf("%s\n", rsJSON)
 			}
-
-			fmt.Println()
-
-			io.Copy(os.Stdout, rs.Output.Stdout())
-			io.Copy(os.Stdout, rs.Output.Stderr())
-
 			fmt.Println()
```

Comment:
We should move this up inside of the text encoding section, and then also make sure we have a newline outside of the loop (it can be printed for all cases that are not text).

This will make it so that all output for a single command will be ld-json, and between each execution will be a newline. This will help break up the output if appending to the same file.

The output should look like if it where being appended to the same file

```
{"Node": 0, ...}
{"Node": 1, ...}
{"Node": 2, ...}

{"Node": 0, ...}
{"Node": 1, ...}
{"Node": 2, ...}
```